### ISSUE TITLE
cannot resolve symbol 'GlideApp' (GlideApp was not generated)

### ISSUE THREAD

**Issue by sonzerolee** [bumptech/glide#1945]
I'm using 4.0.0-RC0, I added it by Gradle:

> repositories {
　mavenCentral() // jcenter() works as well because it pulls from Maven Central
}
dependencies {
　compile 'com.android.support:appcompat-v7:25.3.1'
　compile 'com.github.bumptech.glide:glide:4.0.0-RC0'
　compile 'com.android.support:support-v4:25.3.1'
　annotationProcessor 'com.github.bumptech.glide:compiler:4.0.0-RC0'
}

I followed the here: [https://github.com/bumptech/glide/tree/master/samples/svg/src/main/java/com/bumptech/glide/samples/svg](https://github.com/bumptech/glide/tree/master/samples/svg/src/main/java/com/bumptech/glide/samples/svg)
But when i try load SVG file to ImageView, i get error message: cannot resolve symbol 'GlideApp'
I think GlideApp class was not generated.
My source:

> package com.example.quangson.glidesvg;
import android.graphics.Color;
import android.graphics.drawable.Drawable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import static com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions.withCrossFade;
import android.graphics.drawable.PictureDrawable;
import android.net.Uri;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import com.bumptech.glide.RequestBuilder;
import com.caverock.androidsvg.SVGImageView;
import com.example.quangson.glidesvg.glide.SvgSoftwareLayerSetter;
public class MainActivity extends AppCompatActivity {
　private static final String TAG = "SVGActivity";
　private ImageView imageViewRes;
　private ImageView imageViewNet;
　private RequestBuilder<PictureDrawable> requestBuilder;
　@Override
　protected void onCreate(Bundle savedInstanceState) {
　　super.onCreate(savedInstanceState);
　　setContentView(R.layout.activity_main);
　　imageViewRes = (ImageView) findViewById(R.id.svg_image_view1);
　　imageViewNet = (ImageView) findViewById(R.id.svg_image_view2);
　　requestBuilder = GlideApp.with(this)
　　　　.as(PictureDrawable.class)
　　　　.placeholder(R.drawable.image_loading)
　　　　.error(R.drawable.image_error)
　　　　.transition(withCrossFade())
　　　　.listener(new SvgSoftwareLayerSetter());
　　Uri uri = Uri.parse("http://www.clker.com/cliparts/u/Z/2/b/a/6/android-toy-h.svg");
　　requestBuilder.load(uri).into(imageViewNet);
　}
}

How to generate GlideApp class?
Please help

**Comment by nexodev** [comment]
Any solution ? The same happens to me, adds both glide and annotationProcessor compiler glide

**Comment by sjudd** [comment]
Please comment on your original issue, don't open new identical issues. 

Let me know if following the instructions in #1939 doesn't help.

**Comment by tmm1** [comment]
I have the same issue trying to use `GlideApp` from kotlin.

**Comment by msquare097** [comment]
I am having same issue in Java also.

**Comment by sjudd** [comment]
@niray @tmm1 please file a new issue for kotlin. I'm not especially familiar with the language so your largely on your own. If you do find a solution I'd definitely welcome a contribution.

@msquare097 See the duplicate issue #1939

**Comment by msquare097** [comment]
@sjudd, there is no issue in your library.
I don't know what was the bug, i think there is a problem with android studio.

If there is error some where else in project, then it will point out your bug, while bug is not in your library.
And when I solved that bug it will compile and without any problem.

And to get the confident i repeat this cycle 3 times, and got the same case.

In short when there is error somewhere else in project, it will just point out error in your lib, when we solved that error, then it will run properly.

May be there can be issue with code generation.

**Comment by TWiStErRob** [comment]
It's just how `javac` works, at work we have multiple annotation processors and if we miss an import, or have a typo in variable name, it lists all the generated classes as unresolved symbol from all the different processors (e.g. Dagger 2). The real error message is hidden within tons of unrelated messages, so we need to scroll through all of them to find the problem.

**Comment by trinhvanminh2009** [comment]
@niray It's not working man !

**Comment by timkabot** [comment]
Try to include this in manifest 
<meta-data
            android:name="com.mypackage.MyGlideModule"
            android:value="GlideModule" />

**Comment by timkabot** [comment]
<meta-data
            android:name="com.mypackage.MyGlideModule"
            android:value="GlideModule" />

**Comment by timkabot** [comment]
You need to include in manifest 
<meta-data
            android:name="com.mypackage.MyGlideModule"
            android:value="GlideModule" />

**Comment by timkabot** [comment]
You need to register GlideModule in manifest
https://github.com/bumptech/glide/wiki/Configuration#keeping-a-glidemodule

**Comment by timkabot** [comment]
http://bumptech.github.io/glide/
In glide 4 there is another options

**Comment by PedroOkawa** [comment]
I had the same issue and I found the solution right here on step 2:
http://bumptech.github.io/glide/doc/generatedapi.html

You must create a GlideModule and rebuild:

```
package com.example.myapp;
   
import com.bumptech.glide.annotation.GlideModule;
import com.bumptech.glide.module.AppGlideModule;
   
@GlideModule
public final class MyAppGlideModule extends AppGlideModule {}
```

**Comment by sjudd** [comment]
Duplicate of #1939

**Comment by followthemoney1** [comment]
+1

**Comment by lbx2015** [comment]
+2: I got the same isssue.

**Comment by lbx2015** [comment]
I solved by click Build->ReBuild Project. I don't know what to say this issue follow me several days. 
Could you add these step in the document please, for we all use android studio 2.3.3 for the moment. It worth to do this step description.

**Comment by nixmaldonado** [comment]
File > Invalidate Caches / Restart

**Comment by aamernabi** [comment]
Rebuild project. Thats it..

**Comment by Amit030** [comment]
guys facing the same issue GlideApp variable remains unresolved and when i am using Glide.load()... i am not able to set place holder ..Any of the above mentioned fixes are not working for me.

**Comment by sjudd** [comment]
The generated API is not a requirement, it's just a little bit nicer syntax, see http://bumptech.github.io/glide/doc/options.html and http://bumptech.github.io/glide/doc/migrating.html.

That said, I haven't seen a case where the fixes suggested here and in the documentation don't work. If you've found a case, please attach a sample app that reproduces the problem.

**Comment by wilburt** [comment]
For those experiencing this issue, also make sure `annotationProcessor 'com.github.bumptech.glide:compiler:4.X.X` is added to module level build.gradle

**Comment by boylenssen** [comment]
Also, make sure when using Kotlin, to use `kapt` instead of `annotationProcessor`

kapt 'com.github.bumptech.glide:compiler:4.x.x'

**Comment by vladostaci** [comment]
Finally Solved it. For Kotlin:
- Make sure you're using `kapt "com.github.bumptech.glide:compiler:4.x.x"` instead of `annotationProcessor "com.github.bumptech.glide:compiler:4.6.1"`.
- Check that your Kotlin file inside of which you defined the `@GlideModule` annotated class and the class name is named exactly the same.
Example. The file is named `GlideModule.kt` and it contains the following:
`@GlideModule
class GlideModule : AppGlideModule() {
}`

**Comment by gianglong7695** [comment]
I have the same your problems in version 4.6.1. If you've tried everything and still not succeed like me. You can clone class GlideApp. You need set a new class name.

public final class GlideUtils {
    private GlideUtils() {
    }

    /**
     * @see Glide#getPhotoCacheDir(Context)
     */
    @Nullable
    public static File getPhotoCacheDir(@NonNull Context arg0) {
        return Glide.getPhotoCacheDir(arg0);
    }
    /**
     * @see Glide#getPhotoCacheDir(Context, String)
     */
    @Nullable
    public static File getPhotoCacheDir(@NonNull Context arg0, @NonNull String arg1) {
        return Glide.getPhotoCacheDir(arg0, arg1);
    }

    /**
     * @see Glide#get(Context)
     */
    @NonNull
    public static Glide get(@NonNull Context arg0) {
        return Glide.get(arg0);
    }

    /**
     * @see Glide#init(Glide)
     */
    @Deprecated
    @VisibleForTesting
    @SuppressLint("VisibleForTests")
    public static void init(Glide glide) {
        Glide.init(glide);
    }

    /**
     * @see Glide#init(Context, GlideBuilder)
     */
    @VisibleForTesting
    @SuppressLint("VisibleForTests")
    public static void init(@NonNull Context arg0, @NonNull GlideBuilder arg1) {
        Glide.init(arg0, arg1);
    }

    /**
     * @see Glide#tearDown()
     */
    @VisibleForTesting
    @SuppressLint("VisibleForTests")
    public static void tearDown() {
        Glide.tearDown();
    }

    /**
     * @see Glide#with(Context)
     */
    @NonNull
    public static GlideRequests with(@NonNull Context arg0) {
        return (GlideRequests) Glide.with(arg0);
    }

    /**
     * @see Glide#with(Activity)
     */
    @NonNull
    public static GlideRequests with(@NonNull Activity arg0) {
        return (GlideRequests) Glide.with(arg0);
    }

    /**
     * @see Glide#with(FragmentActivity)
     */
    @NonNull
    public static GlideRequests with(@NonNull FragmentActivity arg0) {
        return (GlideRequests) Glide.with(arg0);
    }

    /**
     * @see Glide#with(Fragment)
     */
    @NonNull
    public static GlideRequests with(@NonNull Fragment arg0) {
        return (GlideRequests) Glide.with(arg0);
    }

    /**
     * @see Glide#with(Fragment)
     */
    @NonNull
    public static GlideRequests with(@NonNull android.support.v4.app.Fragment arg0) {
        return (GlideRequests) Glide.with(arg0);
    }

    /**
     * @see Glide#with(View)
     */
    @NonNull
    public static GlideRequests with(@NonNull View arg0) {
        return (GlideRequests) Glide.with(arg0);
    }
}


Good luck!

**Comment by jemshit** [comment]
Also make sure using `@GlideModule` instead of "@GlideExtension" when extending `AppGlideModule`

**Comment by SinaMN75** [comment]
Well, same happened to me as well, ant it was one of wierdest error I ever got,
I used Butterkife in my project and I found out that if you define the views in private mode, you get this error
```
    @BindView(R.id.tv_option_one)
    TextView tv_option_one;
    @BindView(R.id.tv_option_two)
    TextView tv_option_two;
```
instead of
`    @BindView(R.id.tv_option_one)
    private TextView tv_option_one;
    @BindView(R.id.tv_option_two)
    private TextView tv_option_two;`

**Comment by JorgeAmVF** [comment]
@PedroOkawa's comment solved my issue following what the own documentation states:

> [2. Include a AppGlideModule implementation in your application](http://bumptech.github.io/glide/doc/generatedapi.html#java)

**Comment by paixols** [comment]
**Kotlin**
Followed @vladostaci solution and added these lines to the `proguard-rules.pro` file in the app module: 
-keep public class * extends com.bumptech.glide.module.AppGlideModule
-keep class com.bumptech.glide.GeneratedAppGlideModuleImpl

**Comment by gelbertgel** [comment]
It is not working!!!!!!

**Comment by serafo27** [comment]
I had the same problem.
I notice that Butterknife had problem with glide... i change my listerner from @OnLongClick to @OnClick and now is working.

**Comment by PrinceDholakiya** [comment]
I faced the same problem, And i took solution. Build -> Clean project (After) Rebuild Project.

**Comment by YouCii** [comment]
According to the log provided by AS, we need:

1. apply plugin: 'kotlin-kapt'
2.  kapt 'com.github.bumptech.glide:compiler:x.x.x'

**Comment by Pluckypan** [comment]
make sure:
1. apply plugin: 'kotlin-kapt'
2. kapt 'com.github.bumptech.glide:compiler:x.x.x'

**Comment by c0dehunter** [comment]
Guys please fix this. GlideApp is not generated. We added annotation processor and implemented MyAppGlideModule. Tried rebuilding,c leaning, invalidating caches, ... nothing helps.

**Comment by c0dehunter** [comment]
Found the issue - I needed to first comment out any existing GlideApp calls so project compiled and only after that I could revert the comments and now it's working. But man, so many hoops to jump through :/

**Comment by TWiStErRob** [comment]
@c0dehunter that's how Java's annotation processing works, sadly :(, if there's one error you get all of them reported... it's even worse when you have a project with heavy use of Dagger/Butterknife. A lifesaver to keep in mind: https://stackoverflow.com/a/35707023/253468 so you can find the one that shows the actual error to fix.

**Comment by vishambar555** [comment]
I solved this successfully:

First I created below class in my package:

import com.bumptech.glide.annotation.GlideModule;
import com.bumptech.glide.module.AppGlideModule;
@GlideModule
public class MyAppGlideModule extends AppGlideModule {
}

Then I rebuilt the project by going menu-->build-->rebuild project.

**Comment by muhalifalgibran** [comment]
i'm using kotlin but none of them solve my problem

**Comment by aniketbhoite** [comment]
If you have Kotlin configured in project use kapt 'com.github.bumptech.glide:compiler:x.x.x'

**Comment by byjaka** [comment]
I get the same error 3 times on the same day. I'm trying something out and solving the problem in a ridiculous way to self

**Comment by byjaka** [comment]
I solved the error. This is not the cause of the error, you should look at the error codes should be somewhere else error. I made the mistake wrong because I used Butterknife, so this error came with it.

**Comment by lugondev** [comment]
> Also, make sure when using Kotlin, to use `kapt` instead of `annotationProcessor`
> 
> `kapt 'com.github.bumptech.glide:compiler:4.x.x'`

love you <3

**Comment by RileyManda** [comment]
apply plugin: 'com.android.application'
apply plugin: 'kotlin-android'
apply plugin: 'kotlin-android-extensions'
apply plugin: 'kotlin-kapt'

android{.....
**Add the following dependencies:**
 ```
kapt "android.arch.lifecycle:compiler:1.1.1"
    kapt 'com.github.bumptech.glide:compiler:4.11.0'
    implementation 'com.github.bumptech.glide:glide:4.11.0'
```

**Comment by jenil-vaghani** [comment]
> Please comment on your original issue, don't open new identical issues.
> 
> Let me know if following the instructions in #1939 doesn't help.
> 
> Edit: documentation was updated with Kotlin setup instructions: http://bumptech.github.io/glide/doc/generatedapi.html#kotlin

right brother

**Comment by Ankur-Deka** [comment]
Finally worked, thanks @c0dehunter ! 😀
